### PR TITLE
[JENKINS-37731] Tests for proper handling of SCMTrigger in step.

### PR DESCRIPTION
**Pull request by abayer**
[JENKINS-37731](https://issues.jenkins-ci.org/browse/JENKINS-37731)

Note that we're pretty much stuck with SCMTrigger not working right
with the snippet generator pre-core-2.21 (or whenever
https://github.com/jenkinsci/jenkins/pull/2526 lands). SCMTrigger is
just busted vis a vis the Snippet Generator. So what we're doing here
is testing runtime syntax, and snippet generation against core with a
new enough version to have the "pollScm" symbol on SCMTrigger's descriptor.

Also enabled/fixed some snippetizer round trip tests that now can
work, while I was here.

cc @reviewbybees esp @jglick @daniel-beck

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -74,7 +74,7 @@ THE SOFTWARE.
         <dependency>
             <groupId>org.jenkins-ci.plugins.workflow</groupId>
             <artifactId>workflow-cps</artifactId>
-            <version>2.10</version>
+            <version>2.14-SNAPSHOT</version>
```
Comment (jglick) [inline pom.xml]: What is this depending on?

File: src/test/java/org/jenkinsci/plugins/workflow/multibranch/JobPropertyStepTest.java [hunk 2]
```diff
@@ -336,12 +337,86 @@ public void resetStartsAndStops() {
         assertTrue(p.getTriggers().isEmpty());
 
         // Now add a trigger.
+        // Looking for core versions 2.21 and later for the proper pollScm symbol, rather than the broken scm symbol.
+        if (SCMTrigger.DescriptorImpl.class.isAnnotationPresent(Symbol.class)
+                && SymbolLookup.getSymbolValue(SCMTrigger.DescriptorImpl.class).iterator().next().equals("pollScm")) {
```
Comment (jglick) [inline src/test/java/org/jenkinsci/plugins/workflow/multibranch/JobPropertyStepTest.java]: Would suffice to write

```java
if (SymbolLookup.getSymbolValue(SCMTrigger.class).contains("pollScm")) {
```

right?

File: src/test/java/org/jenkinsci/plugins/workflow/multibranch/JobPropertyStepTest.java [hunk 3]
```diff
@@ -336,12 +337,86 @@ public void resetStartsAndStops() {
...
+
+            p.setDefinition(new CpsFlowDefinition(
+                    "properties([pipelineTriggers([\n"
+                            + "  pollScm(scmpoll_spec: '@daily', ignorePostCommitHooks: true), [$class: 'MockTrigger']])])\n"
```
Comment (jglick) [inline src/test/java/org/jenkinsci/plugins/workflow/multibranch/JobPropertyStepTest.java]: I thought we could use just

```groovy
pollScm('@daily')
```

No?

**Comment by jglick** [comment]
:bee:

File: pom.xml (hunk 1 above)
Comment (abayer) [inline pom.xml]: Honestly, can't remember. Switching to 2.13 to see if that works...

Comment (abayer) [inline pom.xml]: Aaaah, new `structs` version. Adding an explicit one and switching this to 2.13.

File: src/test/java/org/jenkinsci/plugins/workflow/multibranch/JobPropertyStepTest.java (hunk 2 above)
Comment (abayer) [inline src/test/java/org/jenkinsci/plugins/workflow/multibranch/JobPropertyStepTest.java]: Indeed. Good catch.

File: src/test/java/org/jenkinsci/plugins/workflow/multibranch/JobPropertyStepTest.java (hunk 3 above)
Comment (abayer) [inline src/test/java/org/jenkinsci/plugins/workflow/multibranch/JobPropertyStepTest.java]: We can! See the test below, where we are - I wanted to cover both the `pollSCM('@daily')` case and where you want to specify `ignorePostCommitHooks`.

**Comment by abayer** [comment]
Ok, now I remember the reason for cps 2.14-SNAPSHOT - [this and friends](https://jenkins.ci.cloudbees.com/job/plugins/job/workflow-multibranch-plugin/org.jenkins-ci.plugins.workflow$workflow-multibranch/179/testReport/junit/org.jenkinsci.plugins.workflow.multibranch/JobPropertyStepTest/snippetGeneratorOverrideIndexing/) - the fixed snippet generator/`properties` stuff. Removing the structs explicit dependency and switching back to cps 2.14-SNAPSHOT.

**Comment by jglick** [comment]
re:bee:

**Pull request merged**